packages/hydrogen-react/src/storefront-client.ts

```diff
@@ -53,13 +53,22 @@ export function createStorefrontClient(
     );
   }
 
+  const prependProtocol = (domain: string): string =>
+    domain.includes('://') ? domain : `https://${domain}`;
+
   return {
     getShopifyDomain(overrideProps): string {
-      return overrideProps?.storeDomain ?? storeDomain;
+      return prependProtocol(overrideProps?.storeDomain ?? storeDomain);
     },
     getStorefrontApiUrl(overrideProps): string {
-      const finalDomainUrl = overrideProps?.storeDomain ?? storeDomain;
-      return `${finalDomainUrl}${finalDomainUrl.endsWith('/') ? '' : '/'}api/${
+      const finalDomainUrl = prependProtocol(
+        overrideProps?.storeDomain ?? storeDomain,
+      );
```

Comment:
[Nitpick] Can replace this with

```
const getShopifyDomain = (overrideProps): string => {
  const domain = overrideProps?.storeDomain ?? storeDomain;
  return domain.includes('://') ? domain : `https://${domain}`;
}

return {
  getShopifyDomain,
  getStorefrontApiUrl(overrideProps): string {
      const finalDomainUrl = getShopifyDomain(overrideProps)
```

Reply:
I was doing this at first but the types looked a bit weird. Will revert 👍